Games/Solitaire/CardStack.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2020, Till Mayer <till.mayer@web.de>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "Card.h"
+#include <AK/Vector.h>
+
+class CardStack final {
+public:
+    enum Type {
+        Stock,
+        Normal,
+        Waste,
+        Foundation
+    };
+
+    CardStack();
+    CardStack(const Gfx::Point& position, Type location, uint8_t shift_x, uint8_t shift_y, uint8_t step = 1);
+
+    bool is_dirty() const { return m_dirty; }
+    bool is_empty() const { return m_stack.is_empty(); }
+    bool is_focused() const { return m_focused; }
+    Type type() const { return m_type; }
+    size_t count() const { return m_stack.size(); }
+    const Card& peek_const() const { return m_stack.last(); }
+    Card& peek() { return m_stack.last(); }
+    const Gfx::Rect& bounding_box() const { return m_bounding_box; }
+
+    void set_focused(bool focused) { m_focused = focused; }
+    void set_dirty() { m_dirty = true; };
+
+    void push(NonnullRefPtr<Card> card);
+    NonnullRefPtr<Card> pop();
+    void rebound_cards();
+
+    bool check_move(const Card&) const;
+    void add_all_moving(Gfx::Point&, NonnullRefPtrVector<Card>&);
```

Comment:
High-level perspective: What does this API do? It's not self-evident from the declaration.
It writes something to a `Gfx::Point`, and something else to a `NonnullRefPtrVector<Card>&`. Can we name it something that makes it obvious what it does? (Also if it's not intended to write to the `Gfx::Point`, it should be `const` :))